pkgs/development/python-modules/generic/default.nix

```diff
@@ -103,28 +104,38 @@ python.stdenv.mkDerivation (builtins.removeAttrs attrs ["disabled" "doCheck"] //
   # a common idiom in Python
   doInstallCheck = doInstallCheck;
 
+  # We check which test runner is available and run that one
   installCheckPhase = attrs.checkPhase or ''
     runHook preCheck
-    ${python.interpreter} nix_run_setup.py test
+
+    if which py.test; then
+      py.test
+    elif which nosetests; then
+      nosetests
+    else
+      ${python.interpreter} nix_run_setup.py test
+    fi
```

Comment:
I know I sometimes use `py.test` and call it within the `python setup.py test` after some set-up have been done (something like being sure that database schema are deployed). Would not that change break such behavior ? I have no idea how much this kind of setup is used.